### PR TITLE
Add artist and recording MSID checks to dedup

### PR DESCRIPTION
**Pull request by paramsingh**
To allow rows with the same timestamp, we have to add `artist_msid` and `recording_msid` to the influx tags too.

### CODE REVIEW

File: listenbrainz/db/tests/test_lastfm_user.py [hunk 1]
```diff
@@ -63,5 +56,9 @@ def test_user_load_by_id(self):
         self.assertDictEqual(user.__dict__, self.user.__dict__)
 
     def test_user_get_play_count(self):
+        date = datetime(2015, 9, 3, 0, 0, 0)
```
Comment (mayhem) [inline listenbrainz/db/tests/test_lastfm_user.py]: why did you move this block?

File: listenbrainz/listen.py [hunk 2]
```diff
@@ -179,6 +179,8 @@ def to_influx(self, measurement):
             'time' : self.ts_since_epoch,
             'tags' : {
                 'user_name' : escape(self.user_name),
+                'artist_msid' : self.artist_msid,
```
Comment (mayhem) [inline listenbrainz/listen.py]: Why are you using these tags to carry these msids?

Comment (paramsingh) [inline listenbrainz/listen.py]: influx cannot store two rows with the same tags and time, so if we want tracks with the same timestamp but different metadata to be stored, we have to add `artist_msid` and `recording_msid` to tags as well. Otherwise the old listen will just get overwritten by the new one.

File: listenbrainz/db/tests/test_lastfm_user.py (hunk 1 above)
Comment (paramsingh) [inline listenbrainz/db/tests/test_lastfm_user.py]: The test had started failing for me randomly :(

I had added the reason for moving this block to the commit message https://github.com/metabrainz/listenbrainz-server/pull/212/commits/7056af289d1f21107c1b823d4cec27a17f6acdea

File: listenbrainz/listen.py (hunk 2 above)
Comment (mayhem) [inline listenbrainz/listen.py]: Right. Sadly this will bloat our data for little gain, but I can't think of a much better way of doing this.

Comment (mayhem) [inline listenbrainz/listen.py]: Ok, finally coming back to this. Why do we need both artist_msid and recording_msid? Shouldn't just recording_msid be sufficient?

Comment (paramsingh) [inline listenbrainz/listen.py]: Hmm, you're right, `recording_msid` should suffice.

Comment (paramsingh) [inline listenbrainz/listen.py]: Question: do tags not get returned when we do `select count(*) from measurement`? Asking this because I removed them both from fields and added into tags, and the listen count test [here](https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz/listenstore/tests/test_influxlistenstore.py#L197) started to fail because there wasn't a `count_recording_msid` field in the results of the query.

Comment (mayhem) [inline listenbrainz/listen.py]: I believe that is correct. 

So, given all of the above, all fields should be restored as they were and then only recording_msid added to tags. Then we should be ready to merge this -- this now makes sense to me.
